### DISCUSSION TITLE
TransportServer TLS Field

### DISCUSSION THREAD

**Discussion by Seneti42** [nginx/kubernetes-ingress#3754] [Q&A] [answered]
Hi all,

Is anyone having any issues adding the TLS field to the TransportServer custom resource, I keep getting an error that the TLS field isn't in the schema for the k8s.nginx.org/v1alpha1 API even though its in the [documentation](https://docs.nginx.com/nginx-ingress-controller/configuration/transportserver-resource/#tls) and I can see it in the [code](https://github.com/nginxinc/kubernetes-ingress/blob/main/pkg/apis/configuration/v1alpha1/types.go#L82) as well.

I am using the Kubernetes Terraform provider to deploy the TransportServer manifest:
![image](https://user-images.githubusercontent.com/88775596/231483033-045c051a-993b-430d-b79e-b565e41bf958.png)

Without the TLS field in the manifest it deploys fine, any help is appreciated.

**Comment by brianehlert** [comment, marked as answer]
That just came in with 3.1 https://github.com/nginxinc/kubernetes-ingress/pull/3462
What version are you running?
_I am assuming that you installed the CRDs._

**Reply by Seneti42** [reply]
Hi @brianehlert, I'm running 3.1 and yes I have controller.enableCustomResources set to true on the controllers helm chart.

**Reply by Seneti42** [reply]
@brianehlert Thanks for the tip, it made me check if I had the latest CRD's installed, turned out I didn't, I thought I already had them, noob assumptions strike again.

**Reply by brianehlert** [reply]
Glad we could help

**Reply by JyothiAswath** [reply]
Hi, 
I have configured tls in the TransportServer with the secret as given here https://docs.nginx.com/nginx-ingress-controller/configuration/transportserver-resource/#tls however, I am able to send requests from the client without any certificate and the requests are not blocked . I do not think tls is working for me, am I missing any other configurations?

**Reply by JyothiAswath** [reply]
@brianehlert it would be great if you can help with my issue as well as I am blocked

**Reply by brianehlert** [reply]
@JyothiAswath Can you please start a new Discussion, since this one was closed a year ago.
That way we can isolate your issue.

**Comment by jjngx** [comment]
Answered, closing.
